src/memory.js

```diff
@@ -1,18 +1,44 @@
 class MemoryGame {
   constructor(cards) {
     this.cards = cards;
-    // add the rest of the class properties here
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
+   
   }
 
   shuffleCards() {
-    // ... write your code here
+    if (this.cards === undefined) return undefined;
+  
+    let currentIndex = this.cards.length, temporaryValue, randomIndex;
+  
+    
+    while (0 !== currentIndex) {
+  
+      
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex -= 1;
+  
+      
+      temporaryValue = this.cards[currentIndex];
+      this.cards[currentIndex] = this.cards[randomIndex];
+      this.cards[randomIndex] = temporaryValue;
+    }
   }
 
   checkIfPair(card1, card2) {
-    // ... write your code here
+   
+    this.pairsClicked++;
+    if (card1 === card2) {
+      this.pairsGuessed++;
+      return true;
+    }
+    return false;
   }
 
   checkIfFinished() {
-    // ... write your code here
+    
+    if (this.pairsGuessed === this.cards.length / 2) return true;
+    return false;
```

Comment:
Esto te diria que devolvieras directamente la condicion del if, seria lo mismo :D
```suggestion
    return this.pairsGuessed === this.cards.length / 2
```